src/Orleans/Streams/PersistentStreams/StreamQueueBalancerType.cs

```diff
@@ -39,5 +39,10 @@ public enum StreamQueueBalancerType
         /// That is, it does not rebalance queues based on dymanic changes in the cluster Membership.
         /// </summary>
         StaticClusterConfigDeploymentBalancer,
+
+        /// <summary>
+        /// Use user configured custom queue balancer
+        /// </summary>
+        CustomBalancer
```

Comment:
Check out `ReminderTableFactory` for an example of how this is done today. It will have to change a bit for named services, but it's registered in `Silo` as:
```C#
services.AddSingleton<ReminderTableFactory>();
services.AddSingleton<IReminderTable>(sp => sp.GetRequiredService<ReminderTableFactory>().Create());
```

If the user injects their own `IReminderTable`, then that will be used instead.
Ideally we would only add our default one if they have not added their own - maybe @jdom knows a better way?